packages/ts/src/components/donut/modules/arc.ts

```diff
@@ -45,7 +46,7 @@ export function updateArc<Datum> (
 ): void {
   selection
     .style('transition', `fill ${duration}ms`) // Animate color with CSS because we're using CSS-variables
-    .style('fill', (d, i) => getColor(d.data, config.color, i))
+    .style('fill', d => getColor(d.data, config.color, d.index))
```

Comment:
> Shouldn't it be `d.data._index` or will they be the same?

@reb-dev Same as above

Reply:
@rokotyan No actually, because the type `DonutArcDatum<Datum>` expects the `data` property to be the same type as our datum. In the previous code, after getting the arc datum we did this:

```
arcData.forEach((d, i) => {
  d.index = i
  d.innerRadius = innerRadius
  d.outerRadius = outerRadius
})
```

I tweaked it a bit to set d.data to `d.data_datum` and d.index to `d.data_index`. So `DonutDatum<Datum>` exists as a purely intermediate type. However would you prefer I preserve it? We could change the defintition to be `DonutDatum<Datum> extends PieArcDatum<DonuntDatum<Datum>>` ?

Reply:
@reb-dev I see, thanks for explaining.

I did more thinking about it in the morning because the new type inheritance was making the code more difficult to read. Generally, I think that's the right approach. I pushed some changes that will make it slightly clearer in my opinion, i.e. I've added jsdoc comments to the types explaining what `index` means: 
```ts
  ...
  /** Original datum index as in unfiltered data */
  index: number;
```

Let me know if anything there doesn't look right to you. Also feel free to soft-reset the changes and do one single commit. Please don't forget to add a reference to the issue since it's missing in your original commit.

P.S. I've also added a config property for the empty segment size. I think it can be useful in some cases because with certain colors segments can be almost invisible at the default size of 0.5 degrees. Can you please update the docs accordingly?

Reply:
@rokotyan everything looks good, thanks! I just updated the docs and did a single commit.